### DISCUSSION TITLE
Age at Diagnosis - CDE 3225640

### DISCUSSION THREAD

**Discussion by GeoffLyle** [CBIIT/ccdi-federation-api#37] [API Design Discussion]
caDSR Link: [3225640](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=3225640%20and%20ver_nr=2)

Definition: "Age at the time of diagnosis expressed in number of days since birth."

Issues with CDE:
Federation members that collect patient age in years will have to perform a conversion to age in days.  This conversion will not be exact as the reported age in years could be an integer that does not capture the additional age in days.  Since we cannot provide an exact age in days, our reported data will not match the CDE definition.

Agreed Solution as of December 8, 2023:

- Federation does it's own harmonization and does not link to a CDE.  Harmonization to convert age in years to age in days round((365.25 * Age_In_Years)).
- CDEs are being created with definition "approximate age in days".  Federation open API will link to this CDE when it is created.

Other Proposed Solutions:

- Use a CDE that uses ISO 8601 definition.  This will allow for Federation members to report age data in time formats that suit their data.

**Comment by claymcleod** [comment]
I have included this in #49 as a harmonized data element for samples where the element is not backed by a CDE. As you state, future plans include creating a new CDE conforming to an ISO 8601 date the describes the approximate age—I recommend we close this issue and open a new one when the time comes.

**Reply by GeoffLyle** [reply]
Agreed, we can deprecate this data element when a new CDE is created.

**Comment by GeoffLyle** [comment]
Note that the ongoing issues with this data element are similar to https://github.com/CBIIT/ccdi-federation-api/discussions/44 and https://github.com/CBIIT/ccdi-federation-api/discussions/45

When updating this data element remember to those other data elements.

**Comment by GeoffLyle** [comment]
@claymcleod 
Do you remember why we implemented this as a /sample element rather than a /subject element?  Is it to account for secondary malignancies?

**Reply by claymcleod** [reply]
Correct---there can be multiple diagnoses at multiple points in time, so the "age at diagnosis" aligns with sample better. If we were to have a CDE for "Age at _Primary_ Diagnosis", then we discussed that could be aligned to a subject.

**Reply by GeoffLyle** [reply]
From March 14, 2024 meeting:
Discussed that the Federation would like to add a "Age at Primary Diagnosis" field to the `/subject` endpoint.  Would like to work on this after the v1 release.
